pkg/segment/query/processor/statscommand.go

```diff
@@ -58,9 +68,27 @@ func (p *statsProcessor) Process(inputIQR *iqr.IQR) (*iqr.IQR, error) {
 		}
 	}
 
+	// If inputIQR is nil, we are done with the input
+	if inputIQR == nil {
+		iqr := iqr.NewIQR(p.qid)
+		if p.processorType == GroupByCmd {
+			return p.extractGroupByResults(iqr)
+		} else if p.processorType == SegmentStatsCmd {
+			return p.extractSegmentStatsResults(iqr)
+		}
+
+		return nil, toputils.TeeErrorf("qid=%v, statsProcessor.Process: invalid processor type", p.qid)
+	} else if p.finalResultComputed {
+		// Result is already computed but the inputIQR is not nil.
+		// This means, it is the second pass, so there is no need to compute the result again.
+		return nil, nil
```

Comment:
Shouldn't this return `nil, io.EOF`? Also now in `Rewind()` don't you need to update some state indicating you should send the stats again when you get a `nil` input?